core/trino-main/src/main/java/io/trino/server/protocol/Query.java

```diff
@@ -669,7 +668,7 @@ private static Optional<Integer> findCancelableLeafStage(StageInfo stage)
 
         // attempt to find a cancelable sub stage
         // check in reverse order since build side of a join will be later in the list
-        for (StageInfo subStage : Lists.reverse(stage.getSubStages())) {
+        for (StageInfo subStage : stage.getSubStages().reversed()) {
```

Comment:
Aren't these going to be slow?

Preexisting but just asking to understand why this is not considered a performance critical section of code and a manual reverse indexed loop is not used.

Reply:
While it would be slow?

Reply:
I confused these with the non-"view collection" based methods. Nevermind.

Reply:
No I was confused about what the method does. IIRC there is a method in JDK which actually reverses the list instead of just returning a "view"-collection and I was confused about that.

I'm aware that the Guava versions return a view.